### PR TITLE
Stubbed out memory allocation calls in audio library

### PR DESCRIPTION
**Pull request by petewarden**
The micro features library used by the micro_speech example makes direct calls to malloc and new. Since this is used by both Micro and TensorFlow itself, I've added stub calls that direct to either the standard library allocation functions on a desktop OS where a heap is available, or using a simple linear allocation scheme from a fixed pool on embedded systems. This is needed to unblock some library integrations, and is tracked in b/152436073.

### CODE REVIEW

File: tensorflow/lite/experimental/microfrontend/lib/memory_util_fixed_pool.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#include "tensorflow/lite/experimental/microfrontend/lib/memory_util.h"
+
+// This size has been determined by experimentation, based on the largest
+// allocations used by the micro speech example and tests.
+#define FIXED_POOL_SIZE (20 * 1024)
+
+void* microfrontend_alloc(size_t size) {
+  static unsigned char fixed_pool[FIXED_POOL_SIZE];
+  static int fixed_pool_used = 0;
```
Comment (advaitjain) [inline tensorflow/lite/experimental/microfrontend/lib/memory_util_fixed_pool.c:29]: we've had trouble with statics on bluepill. Using a global variable has been our workaround.

https://github.com/tensorflow/tensorflow/blob/8703a34f9d998ff33eae3124bd634e663aed252c/tensorflow/lite/micro/benchmarks/keyword_benchmark.cc#L48-L50

If we can get the micro_speech test to pass on bluepill, I don't care if we use a static or a global.

**Review by advaitjain — CHANGES_REQUESTED** [review]
Can we remove the micro_speech exclusion for the bluepill target?

https://github.com/tensorflow/tensorflow/blob/8703a34f9d998ff33eae3124bd634e663aed252c/tensorflow/lite/micro/tools/make/targets/bluepill_makefile.inc#L72-L74

**Comment by gbaned** [comment]
@petewarden Can you please check @advaitjain's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@petewarden Any update on this PR? Please. Thanks!

File: tensorflow/lite/experimental/microfrontend/lib/memory_util_fixed_pool.c (hunk 1 above)
Comment (petewarden) [reply tensorflow/lite/experimental/microfrontend/lib/memory_util_fixed_pool.c:29]: I've verified this is working (I believe https://github.com/google/stm32_bare_lib/commit/aaabdeb0d6098322a0874b29f6ed547a39b3929f should have helped) so I've left it as a function-level static.

**Review by advaitjain — APPROVED** [review]

**Review by advaitjain — APPROVED** [review]

File: tensorflow/lite/experimental/microfrontend/lib/filterbank_util.c [hunk 2]
```diff
@@ -15,6 +15,7 @@ limitations under the License.
 #include "tensorflow/lite/experimental/microfrontend/lib/filterbank_util.h"
 
 #include <assert.h>
+#include <cstring.h>
```
Comment (advaitjain) [inline tensorflow/lite/experimental/microfrontend/lib/filterbank_util.c:18]: cstring instead of cstring.h

**Review by advaitjain — CHANGES_REQUESTED** [review]

File: tensorflow/lite/experimental/microfrontend/lib/filterbank_util.c (hunk 2 above)
Comment (petewarden) [reply tensorflow/lite/experimental/microfrontend/lib/filterbank_util.c:18]: Doh, thanks!

**Review by advaitjain — APPROVED** [review]

**Comment by bhanuprakashbv** [comment]
@vp-cad

**Pull request merged**